applications/ALEapplication/python_scripts/ale_analysis.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+import KratosMultiphysics.ALEApplication as KratosALE
+
+# Importing the solvers (if available)
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.EigenSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "not imported")
+
+
+class ALEAnalysis(object): # TODO in the future this could derive from a BaseClass in the Core
+    """
+    This class is the main-script of the ALEapplication put in a class
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, ProjectParameters, external_model_part=None):
+        if (type(ProjectParameters) != KratosMultiphysics.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+        self.ProjectParameters = ProjectParameters
+        self.__CreateSolver(external_model_part)
+
+    #### Public functions to run the Analysis ####
+    def Run(self):
+        self.Initialize()
```

Comment:
Is it allowed to call `Run()` more than once? Is `Initialize()` supposed to clear and reset the problem?

Reply:
Actually it would be nice to have a function that resets the model to a state it has after the first ```Initialize()``` call. (Probably also for the structural analysis)